dora/tests/src/test/java/alluxio/client/fs/DoraFileSystemIntegrationTest.java

```diff
@@ -0,0 +1,189 @@
+          .setProperty(PropertyKey.WORKER_HTTP_SERVER_ENABLED, false)
+          .setProperty(PropertyKey.S3A_ACCESS_KEY, mS3Proxy.getAccessKey())
+          .setProperty(PropertyKey.S3A_SECRET_KEY, mS3Proxy.getSecretKey())
+          .setNumWorkers(2)
+          .setStartCluster(false)
+          .build();
+
+  private static final String TEST_BUCKET = "test-bucket";
+  private static final String TEST_FILE = "test-file";
+  private static final AlluxioURI TEST_FILE_URI = new AlluxioURI("/" + "test-file");
+  private static final String TEST_CONTENT = "test-content";
+  private static final String UPDATED_TEST_CONTENT = "updated-test-content";
+
+  private FileSystem mFileSystem = null;
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+  private AmazonS3 mS3Client = null;
+
+  @Before
+  public void before() throws Exception {
+    mLocalAlluxioClusterResource.start();
+    mFileSystem = mLocalAlluxioClusterResource.get().getClient();
+
+    mS3Client = AmazonS3ClientBuilder
+        .standard()
+        .withPathStyleAccessEnabled(true)
+        .withCredentials(
+            new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(mS3Proxy.getAccessKey(), mS3Proxy.getSecretKey())))
+        .withEndpointConfiguration(
+            new AwsClientBuilder.EndpointConfiguration(mS3Proxy.getUri().toString(),
+                Regions.US_WEST_2.getName()))
+        .build();
+    mS3Client.createBucket(TEST_BUCKET);
+  }
+
+  /**
+   * Writes a file through alluxio into UFS. Deletes the file from UFS.
+   * Read the file with sync interval setting to -1 should give the cached file.
+   * Read the file with sync interval setting to 0 should return error.
+   */
+  @Test
+  public void writeThenDeleteFromUfs() throws IOException, AlluxioException {
+    FileOutStream fos = mFileSystem.createFile(TEST_FILE_URI,
+        CreateFilePOptions.newBuilder().setOverwrite(true).build());
+    fos.write(TEST_CONTENT.getBytes());
+    fos.close();
+
+    mS3Client.deleteObject(TEST_BUCKET, TEST_FILE);
+    assertNotNull(mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+        .setCommonOptions(optionNoSync())
+        .build()));
+    try (FileInStream fis = mFileSystem.openFile(TEST_FILE_URI,
+        OpenFilePOptions.newBuilder().setCommonOptions(optionNoSync()).build())) {
+      String content = new String(fis.readAllBytes());
+      assertEquals(TEST_CONTENT, content);
+    }
+
+    assertThrows(FileDoesNotExistException.class, () ->
+        mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+            .setCommonOptions(optionSync()).build()));
+
+    assertThrows(FileDoesNotExistException.class, () ->
+        mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+            .setCommonOptions(optionNoSync()).build()));
+  }
+
+  /**
+   * Writes a file through alluxio into UFS. Deletes the file from UFS.
```

Comment:
s/Deletes the file from UFS/Update the file from UFS (write some new data to the file).